service/history/handler/handler_test.go

```diff
@@ -33,34 +33,44 @@ import (
 	"github.com/stretchr/testify/suite"
 	"go.uber.org/yarpc/yarpcerrors"
 
+	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/cluster"
 	"github.com/uber/cadence/common/log/testlogger"
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/metrics/mocks"
+	"github.com/uber/cadence/common/quotas"
 	"github.com/uber/cadence/common/types"
 	"github.com/uber/cadence/service/history/config"
+	"github.com/uber/cadence/service/history/constants"
 	"github.com/uber/cadence/service/history/engine"
+	"github.com/uber/cadence/service/history/events"
 	"github.com/uber/cadence/service/history/resource"
 	"github.com/uber/cadence/service/history/shard"
+	"github.com/uber/cadence/service/history/task"
 	"github.com/uber/cadence/service/history/workflowcache"
 )
 
 const (
 	testWorkflowID    = "test-workflow-id"
 	testWorkflowRunID = "test-workflow-run-id"
 	testDomainID      = "BF80C53A-ED56-4DD9-84EB-BE9AD4E45867"
+	testValidUUID     = "FCD00931-EBD4-4028-B67E-4DE624641255"
 )
 
 type (
 	handlerSuite struct {
```

Comment:
Also, @taylanisikdemir mentioned that a better practice is _not_ to use test suites because it is better to initiate things like mocks within the smallest range (in each test case).

Reply:
We can refractor the code later. For now, I am just writing tests based on the existing infrastructure.